### ISSUE TITLE
On NHC adjoint

### ISSUE THREAD

**Issue by wwang2** [jakublala/md-neural-ode#1]
Hi

I wanted to inform you that a similar implementation of NHC and Verlet adjoint equations can be found at this GitHub repository: https://github.com/torchmd/mdgrad/blob/cea2332e0ae2c90e2d8caa656c3c51d3b03c59e2/torchmd/sovlers.py#L106. This implementation is licensed under MIT.

Please note that the adjoint equation used in this implementation may induce small gradient errors. Although I am not sure if you are already aware of this issue, it would be nice if someone could fix it at some point. Alternatively, you could use the PyTorch autograd engine for the O(T) backpropagation.

Thank you.

**Comment by jakublala** [comment]
Hi Wujie,

Thanks for commenting on here. This is essentially your implementation copied - sorry for not mentioning it anywhere and not attaching the MIT license. That is all my fault and I apologise.

I didn't know that this implementation may induce small gradient errors! Thank you for notifying me - I will have a look at it once I have the time.
